More Improvements

darxriggs opened the pull request:
I suggest to have a look at the commits individually.

darxriggs committed:
Use lists instead of arrays

A list is more flexible.

darxriggs committed:
Name variable properly

A target is since quite some time not represented by its ip anymore.

darxriggs committed:
Make display name consistent

Most other build steps and post-build actions don't contain a point "." at the end.

darxriggs committed:
Remove unnecessary constructors

darxriggs committed:
Improve logging

- log user friendly information, not the string representation of `Target`
- in case of an invalid URL only log the error message, not the "Publishing data..." message

darxriggs committed:
Remove unused import

darxriggs committed:
Improve formatting

darxriggs committed:
Fix wording

darxriggs committed:
Upgrade configuration-as-code dependency

darxriggs committed:
Upgrade parent pom

darxriggs committed:
Add missed annotation

darxriggs committed:
Add URL to MIT license

darxriggs committed:
Make DescriptorImpl usage more consistent

Now the retrieval of a `Descriptor` instance is similar in `InfluxDbPublisher` and `InfluxDbStep`.

The change requires mocking of the static method `Jenkins#getInstance`.
Issue https://github.com/powermock/powermock/issues/992 was hit along the way.
Therefore the Mockito dependency has to be downgraded slightly.

asimell commented on src/main/java/jenkinsci/plugins/influxdb/InfluxDbStep.java:
```diff
@@ -53,6 +53,7 @@ public String getSelectedTarget() {
         return target;
     }
 
+    @DataBoundSetter
```
Are you sure this is necessary? According to https://jenkins.io/doc/developer/plugin-development/pipeline-integration/ only optional parameters need a `@DataboundSetter`:

> It is a good idea to replace a lengthy `@DataBoundConstructor` with a short one taking just truly mandatory parameters (such as a server location). For all optional parameters, create a public setter marked with `@DataBoundSetter` (with any non-null default value set in the constructor or field initializer).

darxriggs replied:
I think in this case it's not mandatory as the parameter is already part of the constructor. But I think it doesn't hurt either. The goal was to make it consistent with a) the other setters and b) with `InfluxDbPublisher`.

asimell replied:
Actually it's a bug to have it on both the setter and the constructor (ref. https://github.com/jenkinsci/kubernetes-plugin/pull/486), so I removed the `@DataboundSetter` from both InfluxDbStep and InfluxDbPublisher.

asimell commented on pom.xml:
```diff
@@ -194,7 +194,7 @@ THE SOFTWARE.
             <groupId>io.jenkins</groupId>
             <artifactId>configuration-as-code</artifactId>
-            <version>1.14</version>
+            <version>1.31</version>
```
Is there a reason for this dependency upgrade? Most of your other commits have a note about why some version was changed. Same comment applies for parent pom version.

darxriggs replied:
Both upgrades are for staying up-to-date and to receive new features and bug fixes.
I had no specific changes in these dependencies in mind.

darxriggs committed:
Make order of fields and methods consistent

Now each field and method that is both contained in `InfluxDbPublisher`
and `InfluxDbStep` should be sorted the same way in each class.

darxriggs committed:
Remove duplicated Javadoc for fields

It's already documented in `InfluxDbPublicationService`.
Removing it from `InfluxDbPublisher` makes it consisent with `InfluxDbStep`.

darxriggs commented:
I added 2 more commits.

darxriggs commented:
@asimell is there anything I can support?

asimell commented:
@darxriggs Not really, the changes look good. Unfortunately, I partly messed up my dev environment looking into https://issues.jenkins-ci.org/browse/JENKINS-59724, so I haven't been able to run all my test jobs regarding PRs. Sorry for the delay.
